mmv1/products/kms/api.yaml

```diff
@@ -162,6 +162,145 @@ objects:
         'Creating a key':
           'https://cloud.google.com/kms/docs/creating-keys#create_a_key'
       api: 'https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys'
+  - !ruby/object:Api::Resource
+    name: 'CryptoKeyVersion'
+    base_url: '{{crypto_key}}/cryptoKeyVersions'
+    create_url: '{{crypto_key}}/cryptoKeyVersions'
+    self_link: '{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    delete_verb: :POST
+    delete_url: '{{name}}:destroy'
+    description: |
+      A `CryptoKeyVersion` represents an individual cryptographic key, and the associated key material.
+    parameters: 
+      - !ruby/object:Api::Type::String
+        name: 'cryptoKey'
+        required: true
+        input: true
+        url_param_only: true
+        description: |
+          The name of the cryptoKey associated with the CryptoKeyVersions.
+          Format: `'projects/{{project}}/locations/{{location}}/keyRings/{{keyring}}/cryptoKeys/{{cryptoKey}}'`
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource name for this CryptoKeyVersion.
+        output: true
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: |
+          The current state of the CryptoKeyVersion.
+        values:
+          - "PENDING_GENERATION"
+          - "ENABLED"
+          - "DISABLED"
+          - "DESTROYED"
+          - "DESTROY_SCHEDULED"
+          - "PENDING_IMPORT"
+          - "IMPORT_FAILED"
+      - !ruby/object:Api::Type::Enum
+        name: 'protectionLevel'
+        output: true
+        description: |
+          The ProtectionLevel describing how crypto operations are performed with this CryptoKeyVersion.
+        values:
```

Comment:
values for output-only enums aren't used for anything, so this could be removed

Reply:
Sure will do for this and all of the below comments.

Reply:
Removing the enums will cause a compiling error, perhaps if a field is set to Enum, it will by default look for values. I may either remove the field or leave it as it is.

Reply:
I recommend converting it to a String typed field - that should resolve the error and preserve the information.

Reply:
Sounds good, I can do that.